server/utils.go

```diff
@@ -160,3 +160,29 @@ func RemoveDuplicatesFromSlice[T comparable](slice []T) []T {
 	}
 	return list
 }
+
+// SliceStringsMatch checks if two slices contain the same string elements,
+// regardless of order.
+func SliceStringsMatch(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	// create a map to count occurrences of elements in a
+	elementCount := make(map[string]int)
```

Comment:
Nit to ensure enough capacity in one allocation: 
```suggestion
	elementCount := make(map[string]int, len(a))
```